safemysql.class.php

```diff
@@ -170,6 +174,43 @@ public function query()
 		return $this->rawQuery($this->prepareQuery(func_get_args()));
 	}
 
+
+	/**
+	 * Run the provided callback inside a transaction
+	 * @param  callable $callback
+	 * @param  [boolean] $can_retry If the whole transaction should be retried, if it fails due to a deadlock
+	 * @return mixed whatever the callback returns
+	 */
+	public function transaction($callback, $can_retry = false)
+	{
+		if (!empty($this->transactionInProgress))
```

Comment:
Using `!empty()` feels slightly odd to me - I'd've just set `transactionInProgress` to `false` in the constructor, and then been able to do the more straightforward `if ($this->transactionInProgress)` - but whatever.